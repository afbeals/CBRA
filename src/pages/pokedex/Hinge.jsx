// Extenal
import React from 'react';
import PropTypes from 'prop-types';

// Local

// Constants
const classname = 'hinge';

// Component
const Hinge = ({ pClassname, alt }) => (
  <div
    data-testid="pokedex-hinge"
    className={`${pClassname}__${classname}${alt ? ' alt' : ''}`}
  >
    <div className={`${pClassname}__${classname}__cylinder`}>
      <div
        className={`${pClassname}__${classname}__cylinder__section ${pClassname}__${classname}__cylinder__section--top`}
      />
      <div
        className={`${pClassname}__${classname}__cylinder__section ${pClassname}__${classname}__cylinder__section--bottom`}
      />
    </div>
  </div>
);

Hinge.propTypes = {
  pClassname: PropTypes.string.isRequired,
  alt: PropTypes.bool,
};

Hinge.defaultProps = {
  alt: false,
};

export default Hinge;

export { Hinge };
