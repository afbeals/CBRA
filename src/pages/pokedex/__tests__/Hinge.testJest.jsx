// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import { Hinge } from '../Hinge';

// tests
describe('Hinge', () => {
  afterEach(cleanup);
  it('should create Hinge', () => {
    const { getByTestId } = render(<Hinge pClassname={'parent'} />);
    const hinge = getByTestId('pokedex-hinge');
    expect(hinge).toBeTruthy();
    expect(hinge).toMatchSnapshot();
  });

  it('should create alt Hinge', () => {
    const { getByTestId } = render(<Hinge pClassname={'parent'} alt />);
    const atlHinge = getByTestId('pokedex-hinge');
    expect(atlHinge).toBeTruthy();
    expect(atlHinge).toMatchSnapshot();
  });
});
