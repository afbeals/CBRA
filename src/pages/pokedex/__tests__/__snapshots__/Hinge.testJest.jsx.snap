// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hinge should create Hinge 1`] = `
<div
  class="parent__hinge"
  data-testid="pokedex-hinge"
>
  <div
    class="parent__hinge__cylinder"
  >
    <div
      class="parent__hinge__cylinder__section parent__hinge__cylinder__section--top"
    />
    <div
      class="parent__hinge__cylinder__section parent__hinge__cylinder__section--bottom"
    />
  </div>
</div>
`;

exports[`Hinge should create alt Hinge 1`] = `
<div
  class="parent__hinge alt"
  data-testid="pokedex-hinge"
>
  <div
    class="parent__hinge__cylinder"
  >
    <div
      class="parent__hinge__cylinder__section parent__hinge__cylinder__section--top"
    />
    <div
      class="parent__hinge__cylinder__section parent__hinge__cylinder__section--bottom"
    />
  </div>
</div>
`;
